infra/baseServer.go

```diff
@@ -37,17 +27,16 @@ type BaseServer[T IAgent[T]] struct {
 
 func (server *BaseServer[T]) HandleStartOfTurn(iter, round int) {
 	fmt.Printf("Iteration %d, Round %d starting...\n", iter, round)
-	server.beginAgentListeningSession()
 
 }
 
-func (serv *BaseServer[T]) waitForMessagingToEnd() {
+func (serv *BaseServer[T]) endAgentListeningSession() {
+
 	timeoutChannel := time.After(serv.maxMessagingDuration)
 
 agentMessaging:
 	for {
```

Comment:
do we not still have the spinner?

Reply:
this is essentially a manual waitgroup (but the user doesn't have to manage it themselves) that waits for the agents to send their stop talking requests at the end of a round before moving on with the main thread. How I have done it is just a spinner on the server that waits for all agents to finish messaging or a timeout to occur before moving on with the main thread.

Reply:
this is not the same as a spinner on all agent threads though. It just ensures synchronisation of threads between rounds ending and starting

Reply:
From reading online, using a single channel is cleanest for this. Then, you can have both 'normal' writes (representing a `wg.Done()`, achieved in `agentStoppedTalking()`) and 'timed' writes (from the `time.After` channel)

If wg.Wait() never concludes, we get a goroutine leak, which should be avoided